core/src/avm2/activation.rs

```diff
@@ -999,9 +1000,28 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
                 _ => self.unknown_op(op),
             };
 
-            if let Err(e) = result {
-                log::error!("AVM2 error: {}", e);
-                return Err(e);
+            if let Err(error) = result {
+                if let Some(body) = method.body() {
```

Comment:
could this part (after `if let Err(error)`) be extracted to a function, like `fn handle_error`?

Reply:
Done.